models/path.py

```diff
@@ -32,16 +33,20 @@ def link_affected_by_interface(self, interface=None):
                 return link
         return None
 
-    def choose_vlans(self):
+    def choose_vlans(self, controller):
         """Choose the VLANs to be used for the circuit."""
         for link in self:
-            tag = link.get_next_available_tag()
+            tag_value = link.get_next_available_tag(controller, link.id)
+            tag = TAG('vlan', tag_value)
             link.add_metadata("s_vlan", tag)
 
-    def make_vlans_available(self):
+    def make_vlans_available(self, controller):
         """Make the VLANs used in a path available when undeployed."""
         for link in self:
-            link.make_tag_available(link.get_metadata("s_vlan"))
+            tag = link.get_metadata("s_vlan")
+            link.make_tags_available(
```

Comment:
This is ignoring the end result. In normal circumstances we don't expect any surprises here, but if any accidental mistake happens, it'd be great to log as an error and include the rest of the context, just so a bug can be filled and fixed.

Reply:
Added statements to catch any faulty additions.